staging/src/k8s.io/component-base/metrics/http.go

```diff
@@ -61,3 +62,16 @@ func (ho *HandlerOpts) toPromhttpHandlerOpts() promhttp.HandlerOpts {
 func HandlerFor(reg Gatherer, opts HandlerOpts) http.Handler {
 	return promhttp.HandlerFor(reg, opts.toPromhttpHandlerOpts())
 }
+
+// HandlerWithReset return an http.Handler with Reset
+func HandlerWithReset(reg KubeRegistry, opts HandlerOpts) http.Handler {
```

Comment:
It doesn't feel right for this to be just loose in the package. Could it be a method on `reg`?

Reply:
Prometheus decouples these on purpose (I think), because it is possible to reuse a registry with different metric endpoints (and have custom behavior). I would be okay with hanging it off the registry (since I don't we rely off of that behavior in Kubernetes) but I'd feel more comfortable if @brancz ok'd that particular suggestion first.

Reply:
I must not understand what a registry is. Wouldn't all handlers that get passed the same registry show the same data?

Reply:
Technically, it doesn't actually take a registry, it takes a prometheus.Gatherer (which is an interface the Registry satisfies), so you could do things with the registry that would produce a different Gatherer.

Reply:
I see. Except now it takes a KubeRegistry because Gatherer doesn't have a Reset() function? `func HandlerWithReset(reg Gatherer, r Resetter, opts HandlerOpts) http.Handler` as a function signature would make that more obvious.

(no need to change, we've already talked about this longer than it merits)

Reply:
Yes, I've overloaded the Registry a lot already... :(